src/eth/storage/postgres/types.rs

```diff
@@ -14,12 +26,60 @@ pub struct PostgresTransaction {
     pub address_to: Option<Address>,
     pub input: Bytes,
     pub gas: Gas,
+    pub gas_price: Wei,
     pub idx_in_block: Index,
     pub block_number: BlockNumber,
     pub block_hash: Hash,
+    pub result: TransactionExecutionResult,
+    pub block_timestamp: UnixTime,
+    pub output: Bytes,
+    pub value: Wei,
+    pub r: Rs,
+    pub s: Rs,
+    pub v: V,
 }
 
-pub struct PostgresLogs {
+impl PostgresTransaction {
+    pub fn into_transaction_mined(self, logs: Vec<PostgresLog>, topics: Vec<PostgresTopic>) -> TransactionMined {
+        let mined_logs: Vec<LogMined> = logs.iter().map(|log| log.into_log_mined(topics.clone())).collect();
```

Comment:
>Are all mined_logs supposed to contain all topics?

Yes

Reply:
> Also there is a lot of cloning going on
Yes, I did this the more dirty way so I can optimize in the end, that will be fixed soon

Reply:
> maybe you could change PostgresLog::into_log_mined to take self instead of &self and consume the log when doing the conversion, then you could use into_iter() here. You're dropping logs anyway.

Yes, I tried that but ran into other errors and decided to use clone just to have everything working before coming back to optimize

Reply:
> > Are all mined_logs supposed to contain all topics?
> 
> Yes, a topic is what basically defines the log

I understand, but arent you adding all topics to all logs? Wouldn't that make all of the logs in a transaction the same since they are basically defined by the topics and each of the logs contain all of the same topics?